app/src/test/java/org/mozilla/fenix/IntentReceiverActivityTest.kt

```diff
@@ -78,6 +80,24 @@ class IntentReceiverActivityTest {
         assertEquals(true, actualIntent.flags == FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY)
     }
 
+    @Test
+    fun `GIVEN a deeplink intent WHEN processing the intent THEN add the extra flags`() = runBlockingTest {
```

Comment:
nit: Let's improve the name of the test, as now we are not verifying the flags, we are verifying the we get the right class name.

Maybe: 

`GIVEN a deeplink intent WHEN processing the intent THEN add the className HomeActivity`